docs/developer-guide/extensions.md

```diff
@@ -0,0 +1,81 @@
+
+Extensions are plugins that can be distributed as a separate package (a zip file), and be installed, activated and used at runtime.
+Creating an extension is similar to creating a plugin. If you are not familiar with plugins, please, read the [Plugins HowTo page](../plugins-howto) first.
+
+## Introduction
+During this tutorial, you will learn how to create and build a plugin as an extension for MapStore.
+
+## An extension example
+
+A MapStore extension is a plugin, with some additional features.
+
+### web/client/extensions/Extension.jsx
+```javascript
+import {connect} from "react-redux";
+import Extension from "../components/Extension";
+import Rx from "rxjs";
+
+export default {
+    component: connect(state => ({
+        value: state.extension && state.extension.value
+    }), {onIncrease: () => {
+        return {
+            type: 'INCREASE_COUNTER'
+        };
+    }})(Extension),
+    reducers: {
+        extension: (state = {value: 1}, action) => {
+            if (action.type === 'INCREASE_COUNTER') {
+                return {value: state.value + 1};
+            }
+            return state;
+        }
+    },
+    epics: {
+        logCounterValue: (action$, store) => action$.ofType('INCREASE_COUNTER').switchMap(() => {
+            /* eslint-disable */
+            console.log('CURRENT VALUE: ' + store.getState().extension.value);
+            /* eslint-enable */
+            return Rx.Observable.empty();
+        })
+    },
+    containers: {
+        ...
+    }
+};
+```
+As you can see from the code, the most important difference is that you need to export the plugin descriptor **WITHOUT** invoking *createPlugin* on it.
+The extension definition will import or define all the needed dependencies (components, reducers, epics) as well as the plugin configuration elements
+(e.g. containers).
+
+Let's see also the component code, so that we have a working example:
+
+### web/client/components/Extension.jsx
+```javascript
+import React from "react";
+
+const Extension = ({ value = 0, onIncrease }) => {
+    return <div style={{ position: "absolute" }}><span>{value}</span><button onClick={onIncrease}>+</button></div>;
+};
+
+export default Extension;
+```
+
+### Building the compiled extension bundle
+The extension source code has to be stored *INSIDE* the MapStore source code tree. We suggest to use the web/client/extensions folder.
+This is needed so that all the imported dependencies paths are correct.
+
+To build an extension a specific npm run task can be used:
+
+```javascript
+name=myextension version=abc source=extension_source_path npm run build-extension
+```
+
+The parameters that are needed are:
+ - name: the final name prefix of the built bundle
+ - version: will be appended to the name to have a final <name>.<version>.chunk.js filename
+ - source: relative path to the extension jsx file (from the MapStore root folder)
+
+The output js bundle will be written in web/client/dist.
```

Comment:
Two points here:

- [ ] I did it but I didn't find the file in `web/client/dist`

```
name=myextension source=web/client/extensions/MyExtension version=1.0.0 npm run build-extension

> mapstore2@0.1.0 build-extension /home/nali/work/projects/mapstore2/MapStore2
> node --max_old_space_size=2048 ./node_modules/webpack/bin/webpack.js --progress  --colors --config build/extensionBuildConfig.js

 10% building 0/0 modules 0 active(node:13372) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
 10% building 2/3 modules 1 active ...web/client/extensions/MyExtension.jsxBrowserslist: caniuse-lite is outdated. Please run next command `npm update`
Hash: 8aac0aafce92e6dd50de                                                                                                              
Version: webpack 4.36.0
Time: 4278ms
Built at: 2020-01-22 17:12:16
                             Asset       Size               Chunks                    Chunk Names
        myextension.1.0.0.chunk.js  993 bytes          myextension  [emitted]         myextension
vendors~myextension.1.0.0.chunk.js    257 KiB  vendors~myextension  [emitted]  [big]  vendors~myextension
            wrapper.1.0.0.chunk.js   2.38 KiB              wrapper  [emitted]         wrapper
Entrypoint wrapper = wrapper.1.0.0.chunk.js
[./build/pluginWrapper.js] 67 bytes {wrapper} [built]
[./node_modules/react-redux/es/index.js] ./node_modules/react-redux/es/index.js + 34 modules 54.9 KiB {vendors~myextension} [built]
    |    35 modules
[./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {vendors~myextension} [built]
[./node_modules/webpack/buildin/harmony-module.js] (webpack)/buildin/harmony-module.js 573 bytes {vendors~myextension} [built]
[./web/client/extensions/MyExtension.jsx] 1.41 KiB {myextension} [built]
    + 354 hidden modules
ls web/client/dist/ -a
.  ..
```

 - [ ] How to build the zip have to come or can you write it?

Reply:
this tool needs some more work, in any case, the result should be in the plugins folder

Reply:
the build tool needs some more work, in any case the result should be in a /plugins folder